docs/coding-guidelines/project-guidelines.md

```diff
@@ -181,6 +181,6 @@ Each source file should use the following guidelines
 ## Define naming convention
 
 As mentioned in [Conventions for forked code](#conventions-for-forked-code) `#ifdef`ing the code is the last resort as it makes code harder to maintain overtime. If we do need to use `#ifdef`'s we should use the following conventions:
-- Defines based on conventions should be one of `$(OSGroup)`, `$(TargetFramework)`, `$(Configuration)`, or `$(Platform)`, matching exactly by case to ensure consistency.
+- Defines based on conventions should be one of `$(BuildOS)`, `$(TargetFramework)`, `$(Configuration)`, or `$(Platform)`, matching exactly by case to ensure consistency.
```

Comment:
We are using mixture of prefixes here: BuildXXX, TargetXXX, and nothing.

Should we always use `TargetXXX` prefix? Ie. `TargetOS`, `TargetFramework`, ... .

@sdmaclea recently make a number of changes to standardize on `TARGET` prefix in many places, look for `TARGET_UNIX`, `CLR_CMAKE_TARGET_LINUX`, ... .

Reply:
For cross compilation we discussed briefly.  We found the Linux community (gcc & gdb in particular) using: 

+ `Build` - the machine that is running the compiler
+ `Host` - the machine  that will built to execute the compiled code 
+ `Target` - the machine compilation artifact it will debug or generate code

We chose to adopt that convention.

`Build` is typically easily discoverable from ambient properties
`Host` is typically the same as `Target`.  The exception is when we are building cross OS or architecture. CoreClr does this for 
+ Jit testing
+ Cross arch crossgen
+ Cross OS DAC (in progress)
+ `Target` is what we typically want to configure

Reply:
@Anipik @ericstj @ViktorHofer Thoughts about the naming consistency?

Reply:
we can use ```BuildOS``` as it refers to the machine that is running the compiler. The binary built can run on other machine as well.

Reply:
@Anipik that's not correct.  On a Windows machine you can build libraries with BuildOS=Linux which would build the Linux flavor of the libs from a Windows build machine.

I do think `Target` is more consistent here.

Perhaps we should rename `RuntimeOS` to `BuildOS` since that is the property that indicates the build machine's RID.

Reply:
An alternative to `OS` here is `Runtime`  @jkotas what do you think about that?

Reply:
> Perhaps we should rename RuntimeOS to BuildOS since that is the property that indicates the build machine's RID.

+1

> An alternative to OS here is Runtime @jkotas what do you think about that?

Runtime is already overloaded, it could either mean the repository or the runtime, I wouldn't use that one.

Reply:
BuildOS -> TargetOS could be done in this PR. but RuntimeOS -> BuildOS would require an arcade update first to remove the BuildOS from the config system

Reply:
> @Anipik that's not correct. On a Windows machine you can build libraries with BuildOS=Linux which would build the Linux flavor of the libs from a Windows build machine.

What about the coreclr subset ? TargetOS will suggest a wrong meaning there as we cant build Linux version of coreclr on windows machine.

Reply:
> TargetOS will suggest a wrong meaning there as we cant build Linux version of coreclr on windows machine.

You can build the managed parts (System.Private.CoreLib) and I do it all the time. https://github.com/dotnet/runtime/issues/32991#issuecomment-592844413 shows the current command and command to use after Jeremy's build refactoring.

You cannot build the unmanaged portions today because of we do not have the right Windows-hosted Linux-targeting compilers readily available, but that should not be driving our designs.